lib/jekyll-relative-links/generator.rb

```diff
@@ -6,7 +6,7 @@ class Generator < Jekyll::Generator
     include Jekyll::Filters::URLFilters
 
     LINK_TEXT_REGEX = %r!([^\]]+)!
-    FRAGMENT_REGEX = %r!(#.+)?!
+    FRAGMENT_REGEX = %r!(#(?:[^)]+|.+$))?!
```

Comment:
Can you explain what's going on here a bit (so I can debug it if necessary)? 

I see a hash, a non-capturing group that's "one or more character other than `)` or any character to the end of the line, is that right? If so, what's the logic for the `$`?